go-controller/pkg/util/pod_annotation.go

```diff
@@ -116,6 +117,19 @@ func MarshalPodAnnotation(annotations map[string]string, podInfo *PodAnnotation,
 	for _, ip := range podInfo.IPs {
 		pa.IPs = append(pa.IPs, ip.String())
 	}
+
+	existingPa, ok := podNetworks[nadName]
+	if ok {
+		if len(pa.IPs) != len(existingPa.IPs) {
```

Comment:
when can the length not be the same? Isn't dualstack clusters the only use case where we can have two IPs in the pod annotation for the default network?

Reply:
yes, but this is needed to protect the following IP comparision check.